utils/scriptsMongo/actualizarPerfiles.ts

```diff
@@ -0,0 +1,48 @@
+import { AuthUsers } from '../../auth/schemas/authUsers';
+import { Perfiles } from '../../modules/gestor-usuarios/perfil.schema';
+const shiroTrie = require('shiro-trie');
+
+export const actualizarPerfiles = async (done) => {
+    let organizaciones: any = await AuthUsers.aggregate([
+        {
+            $project: {
+                organizaciones: 1,
+                id: '$_id'
+            }
+        },
+        {
+            $unwind: '$organizaciones'
+        }
+    ]);
+    for (let x = 0; x < organizaciones.length; x++) {
+        let usuarioPermisos = shiroTrie.new();
+        usuarioPermisos.add(organizaciones[x].organizaciones.permisos);
+        let perfiles: any = await Perfiles.aggregate([
```

Comment:
Este se puede hacer una sola vez, arriba de todo. El agreggate no tiene sentido. El length del permisos se puede calcular con `perfiles[y].permisos.length`.

Reply:
Lo que decís vos es que en vez de proyectar los datos que uso, haga un find de la colección entera y lo guardo en la variable para después usar??

Reply:
Claor, ahora estas haciendo eso mismo, pero por cada organización. Ese agregate no tiene filtros ni nada por el estilo.

Ademas projectas el tamaño del array pero  tambien esta trayendo a memoria el array así que el tamaño no es necesario si tenes el array entero.

Reply:
claro tenes razón, me quedo adentro del for. Perfecto ahí lo modifico. Gracias!

Reply:
Modificado